examples/with-docker/Dockerfile

```diff
@@ -7,9 +7,9 @@ WORKDIR /app
 # Install dependencies based on the preferred package manager
 COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 RUN \
-  [ -f yarn.lock ] && yarn --frozen-lockfile --prod || \
-  [ -f package-lock.json ] && npm ci || \
-  [ -f pnpm-lock.yaml ] && yarn global add pnpm && pnpm fetch --prod && pnpm i -r --offline --prod || \
+  ([ -f yarn.lock ] && yarn --frozen-lockfile --prod) || \
```

Comment:
This feels less than ideal - maybe we pick one by default (pnpm) and then leave comments for what the other two commands would be for other package managers?

Reply:
Does this supersede https://github.com/vercel/next.js/pull/38693?